### ISSUE TITLE
NAN errors occur at runtime, and all parameters of the proposed network become NAN. 

### ISSUE THREAD

**Issue by mengweiwang** [fregu856/ebms_proposals#1]
I run the source code directly, file '**ebms_proposals/ebm_age/ebm1a_train.py**' can run, file '**ebms_proposals/ebm_age/ebmdn4_train_K4.py**' can not run. 
NAN errors occur at runtime, and all parameters of the proposed network become **NAN**. 
In file '**ebms_proposals/ebm_age/ebmdn4_train_K4.py**' , the root of the problem is line 74: _q_ys_K = torch.exp(q_distr.log_prob(torch.transpose(ys, 1, 0).unsqueeze(2))) # (shape: (1, batch_size, K))_.
The '**q_ys_K**' generated by line 74 is minimal or zero, resulting in **INF** and **NAN** being generated during the calculation of the NCE Loss, and all parameters of the proposed network becoming **NAN** after the gradient update. 
Don't know how you run it.
